### DISCUSSION TITLE
Managing recurring job configuration updates

### DISCUSSION THREAD

**Discussion by jenarros-lw** [jobrunr/jobrunr#660] [General]
We are doing a POC for Jobrunr based on the guidance at https://www.jobrunr.io/en/documentation/background-methods/recurring-jobs/, and we aren't sure how we should setup and update recurring jobs that are shipped with our application.

The docs say: `...for recurring methods to work, at least one BackgroundJobServer should be running all the time and the jobs should be registered on startup of your application`. Given that we have multiple instances of our application. If we register the jobs when the application starts, each instance will try to register the same recurring job. Is this a safe thing to do? or is there a different way we should setup/update the recurring jobs we ship with our application?

Ideally, we would like to update our recurring jobs when we deploy a new application version and delete the ones that are no longer needed.

Thanks in advance for your help

**Comment by rdehuyss** [comment]
Hi @jenarros-lw ,

If you are using any of the framework integrations (Spring Boot / ...), all is well. As a server starts, they just save the same recurring job again. As I think of it, a nice improvement would be that only the master registers the Recurring Job but until now, no one is having issues with this.

The thing I can recommend is giving your Recurring Job's an id, that way you're sure that only one Recurring Job exists as it is otherwise generated via the Job method signature. This way; when you update your jobs, you can refactor your code as long as you keep the id the same.

I hope this makes sense.

All the best,
Ronald

**Reply by jenarros-lw** [reply]
Thanks, @rdehuyss, makes sense.
